core/src/test/scala/shapeless/serialization.scala

```diff
@@ -937,11 +943,11 @@ class SerializationTests {
   def testLazy {
     assertSerializable(Lazy(23))
 
-    assertSerializable(implicitly[Lazy[Generic[Wibble]]])
-    assertSerializable(implicitly[Lazy[Generic1[Box, TC1]]])
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic[Wibble]]])(_.value)
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic1[Box, TC1]]])(_.value)
 
```

Comment:
These are only additional tests. `assertSerializableBeforeAfter` tests that the first argument it is given is serializable upfront (like the former `assertSerializable` did), then it calls its second argument on the first one, and ensure the it is still serializable after that. What was tested before is still tested here.